components/compliance-service/reporting/relaxting/util_test.go

```diff
@@ -1,23 +1,30 @@
 package relaxting
 
 import (
-	"github.com/chef/automate/components/compliance-service/ingest/ingestic/mappings"
-	"github.com/stretchr/testify/assert"
 	"testing"
 	"time"
+
+	"github.com/chef/automate/components/compliance-service/ingest/ingestic/mappings"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestGetEsIndexNoTimeLast24h(t *testing.T) {
 	sum := mappings.IndexNameSum
 	filters := make(map[string][]string)
 
-	index, err := GetEsIndex(filters, true)
-
-	today := time.Now().UTC().Format("2006.01.02")
-	yesterday := time.Now().Add(-24 * time.Hour).UTC().Format("2006.01.02")
+	todayTime := time.Now().UTC()
 
-	assert.Equal(t, sum+"-"+yesterday+"*,"+sum+"-"+today+"*", index)
+	index, err := GetEsIndex(filters, true)
 	assert.NoError(t, err)
+
+	today := todayTime.Format("2006.01.02")
+	if today[len(today)-2:] == "31" {
```

Comment:
@vivek-yadav - I'm working on learning go right now and I can't find this in the docs, so my question is, what effect is the `:` having on the equation in `today[len(today)-2:]`?

Removing the colon gets drastically different numbers, would appreciate if you might be able to explain for point me in the right direction!

Reply:
Here today is string,  when I do today[0] this is char at 0th index of string.
But when we do today[0:1], this returns slice (other way to say array) starting from 0 index to 1st index excluding 1th index.

When I use today[1:], it will return slice starting from 1 to the end of string length.

And here today[len(today)-2:] means this if 
`today = "2006.01.31"`                                                                                   
len(today) = 10
today[10-2:] = today[8:] = start from 8th index upto end of string => today[8] = '3',  today[9] = '1'
thus, today[8:] = "31"

For clear understanding of slices in golang refer this: [slices in golang](https://gobyexample.com/slices)

Reply:
done

Reply:
Thanks so much for this thorough lesson!  I really appreciate it.